### DISCUSSION TITLE
How to disable "Localized Element Sizing"

### DISCUSSION THREAD

**Discussion by Alauddin108** [FloSewn/TQMesh#27] [General]
Hi,
I am a Structural Engineer and trying to use this "TQMesh" library in my applications. In my case, there are sharp corners and I found TQMesh refining the mesh size and number of elements are becoming to big. However, I do not need to refine the mesh size near sharp corners for my particular use. Is it possible to disable "Localized Element Sizing" ? I would appreciate if you can give me some advice about it.

With best regards,

Dr. M Alauddin Biswas
Toronto, Canada.

**Comment by FloSewn** [comment]
Hi Alauddin,

thank you for your interest in this little project.

Could it be that the edges near those sharp corners are very short? Because the element size is primarily estimated from the length of boundary edges. 
A simple way to change this behavior at specific edges, is by adjusting the size and range parameters of the respective vertices at these edges, e.g.: 
```
Vertex& v = domain.add_vertex( x, y, size, range );
```
where `size` controls the local element size and `range` controls the extent around the vertex where this size is applied. 
For your problem, it might be helpful to adjust only the `range` parameter to smaller values. 
This can be achieved by setting the size parameter to zero, e.g.:
```
Vertex& v = domain.add_vertex( x, y, 0.0, range );
```
Now the element size at edges adjacent to this vertex will still be estimated from their lengths (which supports the meshing process) but the elements nearby are allowed to grow faster.


If this doesn't help and you would like to change the actual calculation of the element size, you can have a look at the file `src/algorithm/Domain.h`. 
In that file, the class `SizeFunction` and its function `evaluate()` implements the calculation of the size function. 


Best regards,
Flo

**Comment by Alauddin108** [comment]
Hi FloSewn,
Sorry for delay in responding your reply. Thank you for your time in replying my email with very detail and useful information. You said little project but, I found it very useful, unique and powerful library for meshing.

If you can implement/add "point" and "line" constraint (like in Triangle library) in your "TQMesh" library then it will be more powerful & useful for general use.

I will follow your suggestions to overcome the problems, currently, I am having. I may contact you in the future with more information after trying your suggestions.

With best regards,

Dr. M Alauddin Biswas
Toronto, Canada



________________________________

**Comment by angelmusonda** [comment]
Indeed a feature to define **fixed edges** inside the domain would be very helpful.
